src/nnet3bin/nnet3-compute-batch.cc

```diff
@@ -88,6 +88,7 @@ int main(int argc, char *argv[]) {
     }
 
 #if HAVE_CUDA==1
+    CuDevice::RegisterDeviceOptions(&po);
```

Comment:
This would not work because you are registering the args after calling po.Read(), it would
have to be called further above.
Note that the call CuDevice::RegisterDeviceOptions() does not exist if HAVE_CUDA is not 1.
I think the best solution would be to have the device options class declared outside of the ifdef
(e.g., towards the bottom of cu-device.h), outside of class CuDevice, so that we don't have
to worry about the option being undefined when run without CUDA being compiled in, which
would make sure the script wouldn't crash if that options was given.

Reply:
Why not just put another HAVE_CUDA block and move it before read?  This way it only gets registered if HAVE_CUDA=true?

Reply:
Please see the version I just committed and see if that works well?  If cuda is not used we don't want to still have that option in the command line right?